### PR TITLE
HPCC-13381 Add #option ('obfuscateOutput') to hide original ECL

### PR DESCRIPTION
**Pull request by ghalliday**
Useful if you wish to publish a query to a third part, but hide details
of the ECL implementation.

Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review.  (Compare with ?w=1 to simplify whitespace changes)

File: ecl/eclcc/eclcc.cpp [hunk 1]
```diff
@@ -2248,6 +2248,7 @@ const char * const helpText[] = {
     "! -fexportDependencies    Generate information about inter-definition dependencies",
     "! -fmaxCompileThreads     Number of compiler instances to compile the c++",
     "! -fnoteRecordSizeInGraph Add estimates of record sizes to the graph",
+    "! -fobfuscateOutput       Remove details of the original ECL from output",
```
Comment (jamienoss) [inline ecl/eclcc/eclcc.cpp]: Shouldn't -fshowEclInGraph also be listed?

Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp]: I don't think it is significant enough to include.  This list doesn't include all options - only those that are possibly useful.

Comment (jamienoss) [inline ecl/eclcc/eclcc.cpp]: https://track.hpccsystems.com/browse/HPCC-12313

**Comment by richardkchapman** [comment]
This is a new feature and should not be included in a point release. Please retarget to 5.4

**Comment by ghalliday** [comment]
Closing - see https://github.com/hpcc-systems/HPCC-Platform/pull/7238 instead

**Pull request closed without merging**